app/routes/datasets/versions.py

```diff
@@ -240,13 +240,36 @@ async def append_to_version(
     # For the background task, we only need the new source uri from the request
     input_data = {"creation_options": deepcopy(default_asset.creation_options)}
     input_data["creation_options"]["source_uri"] = request.source_uri
+
+    # If source_driver is "text", this is a datapump request
+    if input_data["creation_options"]["source_driver"] != "text":
```

Comment:
Minor but would be good to use the enum here to make it future-proof `input_data["creation_options"]["source_driver"] != TableDrivers.text`